public/components/AlertInsight/AlertInsight.tsx

```diff
@@ -0,0 +1,211 @@
+ */
+
+import moment from 'moment';
+import { escape } from 'lodash';
+import { OPERATORS_PPL_QUERY_MAP } from '../../pages/CreateMonitor/containers/CreateMonitor/utils/whereFilters';
+import {
+  BUCKET_UNIT_PPL_UNIT_MAP,
+  DEFAULT_ACTIVE_ALERTS_AI_TOP_N,
+  DEFAULT_DSL_QUERY_DATE_FORMAT,
+  DEFAULT_LOG_PATTERN_SAMPLE_SIZE,
+  DEFAULT_LOG_PATTERN_TOP_N,
+  DEFAULT_PPL_QUERY_DATE_FORMAT,
+  PERIOD_END_PLACEHOLDER,
+  PPL_SEARCH_PATH,
+} from '../../pages/Dashboard/utils/constants';
+import { MONITOR_TYPE, SEARCH_TYPE } from '../../utils/constants';
+import { getTime } from '../../pages/MonitorDetails/components/MonitorOverview/utils/getOverviewStats';
+import {
+  filterActiveAlerts,
+  findLongestStringField,
+  searchQuery,
+} from '../../pages/Dashboard/utils/helpers';
+import { getApplication, getAssistantDashboards, getClient } from '../../services';
+import { dataSourceEnabled } from '../../pages/utils/helpers';
+
+export interface AlertInsightProps {
+  alert: any;
+  alertId: string;
+  isAgentConfigured: boolean;
+  children: React.ReactElement;
+  datasourceId?: string;
+}
+
+export const AlertInsight: React.FC<AlertInsightProps> = (props: AlertInsightProps) => {
+  const { alert, children, isAgentConfigured, alertId, datasourceId } = props;
+  const httpClient = getClient();
+  const dataSourceQuery = dataSourceEnabled()
+    ? { query: { dataSourceId: datasourceId || '' } }
+    : undefined;
+
+  const contextProvider = async () => {
+    // 1. get monitor definition
+    const monitorResp = await httpClient.get(
+      `../api/alerting/monitors/${alert.monitor_id}`,
+      dataSourceQuery
+    );
+    const monitorDefinition = monitorResp.resp;
+    // 2. If the monitor is created via visual editor, translate ui_metadata dsl filter to ppl filter
+    let formikToPPLFilters = [];
+    let pplBucketValue = 1;
+    let pplBucketUnitOfTime = 'HOUR';
+    let pplTimeField = '';
+    const isVisualEditorMonitor =
+      monitorDefinition?.ui_metadata?.search?.searchType === SEARCH_TYPE.GRAPH;
+    if (isVisualEditorMonitor) {
+      const uiFilters = monitorDefinition?.ui_metadata?.search?.filters || [];
+      formikToPPLFilters = uiFilters.map((filter) =>
+        OPERATORS_PPL_QUERY_MAP[filter.operator].query(filter)
+      );
+      pplBucketValue = monitorDefinition?.ui_metadata?.search?.bucketValue || 1;
+      pplBucketUnitOfTime =
+        BUCKET_UNIT_PPL_UNIT_MAP[monitorDefinition?.ui_metadata?.search?.bucketUnitOfTime] ||
+        'HOUR';
+      pplTimeField = monitorDefinition?.ui_metadata?.search?.timeField;
+    }
+    delete monitorDefinition.ui_metadata;
+    delete monitorDefinition.data_sources;
+
+    // 3. get data triggers the alert and fetch log patterns
+    let monitorDefinitionStr = JSON.stringify(monitorDefinition);
+    let alertTriggeredByValue = '';
+    let dsl = '';
+    let index = '';
+    let topNLogPatternData = '';
+    if (
+      monitorResp.resp.monitor_type === MONITOR_TYPE.QUERY_LEVEL ||
```

Comment:
For my own understanding, what's the process for other monitor types?

Reply:
Other monitor types like `per cluster metrics` don't have a query dsl associated with it, so we don't have the process to query data.

we may have a follow up to review `Per document` and `Composite monitor` monitor types